### DISCUSSION TITLE
Fetching Athena S3 output CSV instead of iterating Queryresult will boost loading data

### DISCUSSION THREAD

**Discussion by jgrumboe** [cloud-carbon-footprint/cloud-carbon-footprint#1042] [Feature suggestions]
## Feature Suggestion
**Is your feature request related to a problem? Please describe.**
Fetching billing data for multiple days/months via AWS Athena is extremely slow. The reason is that Athena QueryResult API is only fetching a maximum of 1000 records per invocation. It just scales enough for monthly grouping but not for daily.

**Describe the solution you'd like**
I found this post https://stackoverflow.com/a/64031488, which suggest fetching the S3 output CSV directly, instead of iterating over QueryResult API.

**Additional context**
https://stackoverflow.com/a/64031488

**Comment by ccasher** [comment]
Hello Johannes, thanks for the new issue! Are you comfortable at all working in TS/JS? If so, would you want to start working on this implementation and we could help collaborate? Otherwise, this sounds like a great idea so we could add it to our backlog.

**Comment by jgrumboe** [comment]
Honestly, I'm good at copy/paste but not in generating new code in TS/JS ☺

So, add it to the backlog, please.

**Comment by ccasher** [comment]
Sounds good, thank you!
